include/rocksdb/advanced_options.h

```diff
@@ -773,6 +774,30 @@ struct AdvancedColumnFamilyOptions {
   // Dynamically changeable through the SetOptions() API
   CompressionType blob_compression_type = kNoCompression;
 
+  // UNDER CONSTRUCTION -- DO NOT USE
+  // Enables garbage collection of blobs. Blob GC is performed as part of
+  // compaction. Valid blobs residing in blob files older than a cutoff get
+  // relocated to new files as they are encountered during compaction, which
+  // makes it possible to clean up blob files once they contain nothing but
+  // obsolete/garbage blobs. See also blob_garbage_collection_cutoff below.
+  //
+  // Default: false
+  //
+  // Dynamically changeable through the SetOptions() API
+  bool enable_blob_garbage_collection = false;
```

Comment:
I remember there was a discussion about naming such options as adjectives, e.g. `blob_garbage_collection_enabled`?

Reply:
What is the expected behavior if this options is changed from true to false during a compaction that has already started?

Reply:
> I remember there was a discussion about naming such options as adjectives, e.g. `blob_garbage_collection_enabled`?

Hm, I'm not aware of that discussion but the prevailing convention is definitely the imperative style, e.g.:

```
options.h:  bool enable_thread_tracking = false;
options.h:  bool enable_pipelined_write = false;
options.h:  bool enable_write_thread_adaptive_yield = true;
table.h:  bool enable_index_compression = true;
options.h:  bool disable_auto_compactions = false;
options.h:  bool disableWAL;
```

I'd also rather keep it in sync with `enable_blob_files` (which would be a bit of a pain to change at this point since it's been part of the public API for some time).

Reply:
> What is the expected behavior if this options is changed from true to false during a compaction that has already started?

Since `MutableCFOptions` is copied to a member in `Compaction`, in-progress compactions are never affected by `SetOptions`; they always use the same saved values throughout the compaction. This is true in general; I can call it out in a note for these options though.

Reply:
> Since `MutableCFOptions` is copied to a member in `Compaction`
>
Thanks for clarification.